src/feature/nodelist/microdesc.c

```diff
@@ -221,6 +221,13 @@ dump_microdescriptor(int fd, microdesc_t *md, size_t *annotation_len_out)
   }
 
   md->off = tor_fd_getpos(fd);
+  const char *nulpos = memchr(md->body, 0, md->bodylen);
+  if (BUG(nulpos)) {
+    log_warn(LD_BUG, "About to dump a NUL into a microdescriptor file. "
+             "offset %"PRId64", bodylen %zu, nul position %zu",
```

Comment:
Windows says:
"error: unknown conversion type character 'z' in format"
https://ci.appveyor.com/project/torproject/tor/builds/23576412/job/80p2k2cumplatqx9#L1468